api/main_endpoints/routes/User.js

```diff
@@ -199,6 +206,33 @@ router.post('/getPagesPrintedCount', (req, res) => {
   });
 });
 
+router.get('/callback', async function(req, res) {
+  const code = req.query.code;
+  const email = req.query.state;
+  const verified = discordConnection.loginWithDiscord(code, email);
+  if (verified) {
+    return res.status(OK).redirect('https://discord.com/oauth2/authorized');
+  }
+  return res.status(NOT_FOUND).send('Authorization unsuccessful!');
+});
+
+router.post('/connectToDiscord', function(req, res) {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+  const email = req.body.email;
```

Comment:
i would put this above `checkIfTokenSent` on line 220. seems like theres no point of checking token stuff if theres no email to begin with